### ISSUE TITLE
Trial is completely useless

### ISSUE THREAD

**Issue by hugoeustaquio** [vladmihalcea/hypersistence-optimizer#101]
Hello, I'm a java developer and I've read about hypersistence-optimizer. I wanted to introduce it to my coworkers. It happens that I can't justify the investiment of buying a new software with the following output:

```

2020-06-20 20:19:15,181 [ThreadId: restartedMain] ERROR Hypersistence Optimizer                        - CRITICAL - Fetch1Event - The optimization event description is hidden in the Trial Version. If you want to know what performance-related problem has triggered this event, you need to purchase the product.
2020-06-20 20:19:15,188 [ThreadId: restartedMain] WARN  Hypersistence Optimizer                        - The Hibernate SessionFactory is not decorated by Hypersistence Optimizer, and, for this reason, the runtime checks will be disabled.
2020-06-20 20:19:15,190 [ThreadId: restartedMain] WARN  Hypersistence Optimizer                        - 688 issues were found: 0 BLOCKER, 468 CRITICAL, 152 MAJOR, 68 MINOR

```

The trial version don't show absolutely nothing relevant. It demonstrate zero possible performance gains. How can I try to convince somewone to "try" a product if this "trial version" have absolutelly nothing?

Is there a way to know some limited functionality with the trial version?

**Comment by vladmihalcea** [comment]
> How can I try to convince someone to "try" a product if this "trial version" have absolutely nothing?

The trial isn't useless at all. In fact, it tells you that you have 688 problems that could be fixed if you bought the product. I bet your team would like to get those 688 problems fixed, right?

When I released the 1.0 version of the product, the trial version had all the functionality. However, during the first 3 months. So, what happened during that time? People got the product, fixed their problems, and didn't need to buy it.

In June 2019, I created a new trail branch that obfuscated the optimization tips. Guess what happened? People started buying it because,  now, they needed it to do so.

That's why the trial version only tells you the number of problems and the type of problems that were identified: configuration, mapping, runtime.

> Is there a way to know some limited functionality with the trial version?

The product comes with a 30-day money-back guarantee. So, you can safely buy it, and if you are not happy, you'll get your money back. So far, the only time I had to give a refund was to someone who bought the tool to optimize an EclipseLink-based product, and, since the tool only supports Hibernate, the refund allowed them to get all the money back.

**Comment by shaharco** [comment]
> ### The new Trial Version addresses this issue
> Starting with the 2.2 version, the Trial version now reveals **5** Mapping and **5** Configuration issues so that it's now easier for you to find out how cool this library is.
> 
> Now, to answer your questions:
> 
> > So, how could I convince my manager to buy the product if this trial version doesn't have the same functionality as the full version?
> 
> The trial version has just told you you have **688** problems that could be fixed if you bought the product.
> 
> > 688 issues were found: 0 BLOCKER, 468 CRITICAL, 152 MAJOR, 68 MINOR
> 
> If you tell your manager that Hypersistence Optimizer can help you fix **688** issues, I think this a great reason to consider buying a license.
> 
> ### A history lesson
> When I released the 1.0 version of the product, the trial version had all the functionality as the Full Version. I thought that people would surely buy the product after they found how many issues they could get fixed by using Hypersistence Optimizer.
> 
> But, I was wrong!
> 
> In fact, this is what happened. Because the Trial Version allowed people to fix their performance problems for free, a single person decided that it was fair to buy it in the first 3 months. The rest chose to fix their application performance issues and moved on. If a service can be obtained free of charge, there is no incentive to buy it. That's also the reason [why donations rarely work for an open-source project](https://twitter.com/vlad_mihalcea/status/1194584719133958146).
> 
> In June 2019, I created a new branch for the trial version where I hid the optimization tips. And, all of a sudden, a magical thing happened. People started buying it. One year later, and the product has 221 customers.
> 
> That's why the trial version only tells you the number of problems and the generic type of problem that were identified: configuration, mapping, runtime.
> 
> For instance, you got the following message:
> 
> ```
> 2020-06-20 20:19:15,181 [ThreadId: restartedMain] ERROR Hypersistence Optimizer - CRITICAL - Fetch1Event - The optimization event description is hidden in the Trial Version. If you want to know what performance-related problem has triggered this event, you need to purchase the product.
> ```
> 
> You have a fetch-related issue that affects the performance of your system. After you buy the product, not only you will know what exactly the problem is, but the documentation will tell you how to fix it as well.
> 
> ### How can I convince my manager to buy a license?
> > Is there a way to get access to the full version features when using the trial version?
> 
> The product comes with a 30-day money-back guarantee. So, you can safely buy it, and if you are not happy, you'll get your money back. So far, the only time I had to give a refund was to someone who bought the tool to optimize an EclipseLink-based product, and, since the tool only supports Hibernate, the refund allowed them to get all the money back.
> 
> #### One of the greatest advantages of buying a Hypersistence Optimizer license
> A great advantage of using Hypersistence Optimizer is that [you could make the build fail](https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#How_to_trigger_a_build_failure_when_detecting_any_performance_issue) when any issue is detected.
> 
> This way, you will be notified of future issues at the moment the problems get introduced, so you can fix them right away. This way, you can make sure the production system will never be affected by those issues.
> 
> That's a very compelling reason for your team to consider using Hypersistence Optimizer.
> 
> #### Watch a video presentation demonstrating Hypersistence Optimizer capabilities
> A great way to convince your manager or your team about the benefits of using Hypersistence Optimizer is to watch the following video on YouTube which shows how easily you can spot JPA and Hibernate issues in a Spring Boot application:
> 
> [<img alt="" width="100%" src="https://camo.githubusercontent.com/cb417454d1e9603120b97acdf3ea7a6fcba772ae/68747470733a2f2f696d672e796f75747562652e636f6d2f76692f78316e4f566374395032672f6d617872657364656661756c742e6a7067">](https://youtu.be/x1nOVct9P2g)

@vladmihalcea i'm trying trial version 2.2.0 but still everything is hidden, i don't see the 10 issues you say are revealed in the trial version.

**Comment by vladmihalcea** [comment]
The Trial Version allows you to find the following issues:

- [`ConsoleStatementLoggingEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#ConsoleStatementLoggingEvent)
- [`EagerFetchingEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#EagerFetchingEvent)
- [`ElementCollectionEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#ElementCollectionEvent)
- [`JdbcBatchOrderInsertsEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#JdbcBatchOrderInsertsEvent)
- [`JdbcBatchOrderUpdatesEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#JdbcBatchOrderUpdatesEvent)
- [`JdbcBatchSizeEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#JdbcBatchSizeEvent)
- [`ManyToManyCascadeRemoveEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#ManyToManyCascadeRemoveEvent)
- [`OneToOneWithoutMapsIdEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#OneToOneWithoutMapsIdEvent)
- [`SchemaGenerationEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#SchemaGenerationEvent)
- [`UnidirectionalOneToManyEvent`]https://vladmihalcea.com/hypersistence-optimizer/docs/user-guide/#UnidirectionalOneToManyEvent)

If you don't see any of those, but just the other obfuscated ones, it means you are not affected by the ones which were chosen for the Trial Version.

Just make a `fetch` attribute `FetchTYpe.LAZY` or use a `@OneToOne` without `@MapsId` or a unidirectional `@OneToMany`, and you will see the issues being reported.
